src/main/java/innohack/gem/service/extract/TikaContentExtractor.java

```diff
@@ -0,0 +1,70 @@
+package innohack.gem.service.extract;
+
+import com.beust.jcommander.internal.Lists;
+import innohack.gem.entity.GEMFile;
+import innohack.gem.entity.extractor.ExtractConfig;
+import innohack.gem.entity.extractor.ExtractedRecords;
+import innohack.gem.entity.extractor.TimestampColumn;
+import innohack.gem.entity.feature.AbstractFeature;
+import innohack.gem.entity.feature.CsvFeature;
+import innohack.gem.entity.feature.TikaFeature;
+import innohack.gem.entity.rule.ParamType;
+import innohack.gem.entity.rule.Parameter;
+import java.text.ParseException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class TikaContentExtractor extends AbstractExtractor {
+
+  public TikaContentExtractor() {
+    this(null);
+  }
+
+  public TikaContentExtractor(String regex) {
+    setParams(Lists.newArrayList(new Parameter("Content", "", ParamType.STRING, regex)));
+  }
+
+  @Override
+  public ExtractedRecords extract(GEMFile f, ExtractConfig extractConfig) throws Exception {
+    f.extract();
+    for (AbstractFeature feature : f.getData()) {
+      if (feature instanceof TikaFeature) {
+        String regexString = getParams().get(0).getValue();
+
+        ExtractedRecords results = new ExtractedRecords();
+        String tikaContent = ((TikaFeature) feature).getContent();
+
+        Pattern pattern = Pattern.compile(regexString, Pattern.CASE_INSENSITIVE);
+        Matcher matcher = pattern.matcher(tikaContent);
+        int count = 0;
+
+        List<String> foundResults = Lists.newArrayList();
+        while (matcher.find()) {
+          count++;
+          String matchText = tikaContent.substring(matcher.start(), matcher.end());
```

Comment:
Are we assuming no headers for this extraction? Or will it be an optional param specified by user? (if there's header, first row is header?)

Reply:
this one i also lost liao.. for tikaextraction is meant for pure textual and not only tabular content?

Reply:
I also dunno... I am inferring from the UI... so we'll just assume extract blobs of text then. In future, can extend to extract blobs of text, table with headers and tables without headers (via more params)